### PR TITLE
When resetting with no selection, apply reset to current line.

### LINKED ISSUE #23: FR: if no line is selected, `Reset selected checklists` falls back to applying to current line
For convenience, if there is no line selected, it would be nice if the current line were considered selected as a fallback. This is a very common case.

### PR DESCRIPTION
**Pull request by huyz**
Resolves https://github.com/lhansford/obsidian-checklist-reset/issues/23

## Pull request overview

This PR adds a small, testable helper to normalize Markdown editor selections for the "Reset selected checklists" command, and adds unit tests to ensure collapsed selections are treated as "current line" (instead of doing nothing).

**Changes:**

• Extract getMarkdownSelectionToReset helper to normalize selection vs. cursor state.
• Added restoreCursorAfterReset helper to encapsulate cursor restoration rules and line-length clamping.
• Add unit tests covering non-empty selections and collapsed-selection fallback-to-line behavior and for cursor clamping after content shrink and for skipping restoration when multiple selections are active
• Wire main.ts to use the new helpers when invoking handleMarkdownAction for Markdown views.

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -108,12 +110,17 @@ export default class ChecklistReset extends Plugin {
         } else if (view instanceof MarkdownView) {
           const selectedText = view.editor.listSelections();
           if (selectedText.length > 0) {
+            const selection = selectedText[0];
```
[the remark this answers is not in the document]
Comment (huyz) [reply main.ts:113]: That makes sense. But you're referring to code written by the upstream maintainer, not me. So, out of scope

File: main.ts [hunk 2]
```diff
@@ -108,12 +110,17 @@ export default class ChecklistReset extends Plugin {
...
+            const lineLength = view.editor.getLine(selection.head.line).length;
+
             handleMarkdownAction(
               view,
               this.settings,
               "uncheck",
-              selectedText[0]
+              getMarkdownSelectionToReset(selection, lineLength)
             );
```
[the remark this answers is not in the document]
Comment (huyz) [reply main.ts:121]: This computation is trivial. Not worth complicating the code.

File: src/getMarkdownSelectionToReset.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import type { EditorSelection } from "obsidian";
+
+export function getMarkdownSelectionToReset(
+  selection: EditorSelection,
+  lineLength: number
+): EditorSelection {
```
[the remark this answers is not in the document]
Comment (huyz) [reply src/getMarkdownSelectionToReset.ts:6]: This isn't an API, just an internal method. Not worth changing.
